src/gpu/kmeans/KmMatrix/KmConfig.h

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright 2018 H2O.ai, Inc.
+ * License   Apache License Version 2.0 (see LICENSE for details)
+ */
+
+#ifndef KM_CONFIG_H_
+#define KM_CONFIG_H_
+
+#define USE_CUDA() 1
+
+#include "stdio.h"
+
+// Matrix host dev
+#define M_HOSTDEV __host__ __device__
+#define M_DEV __device__
+#define M_DEVINLINE __device__ __forceinline__
+#define M_HOSTDEVINLINE __host__ __device__ __forceinline__
+
```

Comment:
This cuda check macro is duplicated here: https://github.com/h2oai/h2o4gpu/blob/2f849e24279a31482d56b8f6c51232bb399bd7aa/src/gpu/kmeans/kmeans_general.h#L13

And here:
https://github.com/h2oai/h2o4gpu/blob/3eb471579438e14cd62a62243b46ba1ea494dd45/src/gpu/utils/utils.cuh#L36

I suggest you use the one in utils and delete the one in kmeans. My personal preference is to use the variant of your checking that throws an exception. This gives you more control of how you want to program to exit. We should use the same method throughout H2O4GPU.

Reply:
I intend to remove all these utils code in KmMatrix later. Sorry for not stating it out. It's just I don't want to pollute the rest of the repo at this stage. And the namespace should be consistent with rest of the project, though I haven't figure out how to do this yet.